pkg/specgen/generate/pause_image.go

```diff
@@ -33,64 +32,28 @@ func PullOrBuildInfraImage(rt *libpod.Runtime, imageName string) (string, error)
 		return imageName, nil
 	}
 
-	name, err := buildPauseImage(rt, rtConfig)
-	if err != nil {
-		return "", fmt.Errorf("building local pause image: %w", err)
-	}
-	return name, nil
+	return "", nil
 }
 
-func buildPauseImage(rt *libpod.Runtime, rtConfig *config.Config) (string, error) {
-	version, err := define.GetVersion()
-	if err != nil {
-		return "", err
-	}
-	imageName := fmt.Sprintf("localhost/podman-pause:%s-%d", version.Version, version.Built)
-
-	// First check if the image has already been built.
-	if _, _, err := rt.LibimageRuntime().LookupImage(imageName, nil); err == nil {
-		return imageName, nil
-	}
-
+func PrepareInfraSpec(rtConfig *config.Config, infraCommand []string, infraContainerSpec *specgen.SpecGenerator) error {
 	// Also look into the path as some distributions install catatonit in
 	// /usr/bin.
 	catatonitPath, err := rtConfig.FindInitBinary()
 	if err != nil {
-		return "", fmt.Errorf("finding pause binary: %w", err)
+		return fmt.Errorf("finding pause binary: %w", err)
 	}
-
-	buildContent := fmt.Sprintf(`FROM scratch
-COPY %s /catatonit
-ENTRYPOINT ["/catatonit", "-P"]`, catatonitPath)
-
-	tmpF, err := os.CreateTemp("", "pause.containerfile")
+	catatonitPath, err = filepath.EvalSymlinks(catatonitPath)
```

Comment:
I don't think we should use the catatonit dir directly. We don't know what other stuff will be in the directory.

My personal preference would be to create a an empty tmpd dir and use an overlay on top and then bind mount just the catatonit binary.
However there is a larger problem with this. We need to but some of this logic into libpod. pods can stay around for a while and the libpod config is immutable. If the catatonit location on the host would ever change this will get broken. And while on a regular distro that would be weird it will happen easily on a distro like NixOS.

cc @mheon

Reply:
Concur, this logic needs to live in libpod. I believe pause containers already have a bool in container config, which we could use to activate alternative storage logic to make a directory (I don't think an overlay or temporary directory are necessary, pod pause containers should always be read-only rootfs so no need to have an overlay). The difficulty would be that we would still need to respect alternative, user-specified infra images, so just checking for a pause container can't unconditionally be used to decide on this alternative logic.

Reply:
I've discussed this with @Luap99 and I will so far use the overlay, but move the code to libpod. Using the read-only FS without could be additional RFE done in follow-up PR if that's fine by others in this discussion.

Reply:
```
/tmp/tmp.VO8Dj7fyoa
├── catatonit
├── dev
├── etc
│   ├── hostname
│   ├── hosts
│   ├── mtab -> /proc/mounts
│   └── resolv.conf
├── proc
├── run
│   └── secrets
└── sys
```

These are the current default inodes we would have to precreate if we want to go read only. I don't see a huge gain for that and it would complicate the logic futher as we might have to deal with default containers.conf mount points as well.
As such the overlay option makes the implementation much simpler for now which I think is better.